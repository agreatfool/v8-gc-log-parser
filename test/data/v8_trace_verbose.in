[23252:0x102801600]      769 ms: Scavenge 11.8 (18.5) -> 8.0 (18.5) MB, 0.5 / 0.0 ms  allocation failure
[23252:0x102801600] Memory allocator,   used:  18944 KB, available: 1447424 KB
[23252:0x102801600] New space,          used:     90 KB, available:   3937 KB, committed:   8192 KB
[23252:0x102801600] Old space,          used:   4983 KB, available:      0 KB, committed:   5120 KB
[23252:0x102801600] Code space,         used:   2523 KB, available:      3 KB, committed:   3072KB
[23252:0x102801600] Map space,          used:    621 KB, available:      0 KB, committed:   1024 KB
[23252:0x102801600] Large object space, used:      0 KB, available: 1446903 KB, committed:      0 KB
[23252:0x102801600] All spaces,         used:   8218 KB, available: 1450845 KB, committed:  17408KB
[23252:0x102801600] External memory reported:     56 KB
[23252:0x102801600] Total time spent in GC  : 7.0 ms
[23252:0x102801600] Fast promotion mode: false survival rate: 3%
[23252:0x102801600]      921 ms: Scavenge 12.0 (19.0) -> 8.2 (19.0) MB, 0.6 / 0.0 ms  allocation failure
[23252:0x102801600] Memory allocator,   used:  19456 KB, available: 1446912 KB
[23252:0x102801600] New space,          used:     47 KB, available:   3980 KB, committed:   8192 KB
[23252:0x102801600] Old space,          used:   5110 KB, available:    375 KB, committed:   5632 KB
[23252:0x102801600] Code space,         used:   2534 KB, available:      3 KB, committed:   3072KB
[23252:0x102801600] Map space,          used:    677 KB, available:      0 KB, committed:   1024 KB
[23252:0x102801600] Large object space, used:      0 KB, available: 1446391 KB, committed:      0 KB
[23252:0x102801600] All spaces,         used:   8369 KB, available: 1450751 KB, committed:  17920KB
[23252:0x102801600] External memory reported:     58 KB
[23252:0x102801600] Total time spent in GC  : 7.6 ms
[23252:0x102801600] Fast promotion mode: false survival rate: 4%
[23252:0x102801600]      930 ms: Heap growing factor 4.0 based on mu=0.970, speed_ratio=0 (gc=0, mutator=7009)
[23252:0x102801600]      930 ms: Grow: old size: 6813 KB, new limit: 31280 KB (4.0)
[23252:0x102801600]      930 ms: Mark-sweep 8.4 (19.0) -> 6.8 (19.0) MB, 0.8 / 0.0 ms  (+ 6.2 ms in 10 steps since start of marking, biggest step 1.0 ms, walltime since start of marking 14 ms) finalize incremental marking via task GC in old space requested
[23252:0x102801600] Memory allocator,   used:  19456 KB, available: 1446912 KB
[23252:0x102801600] New space,          used:    145 KB, available:   3882 KB, committed:   8192 KB
[23252:0x102801600] Old space,          used:   4233 KB, available:    469 KB, committed:   5632 KB
[23252:0x102801600] Code space,         used:   2287 KB, available:      0 KB, committed:   3072KB
[23252:0x102801600] Map space,          used:    292 KB, available:      0 KB, committed:   1024 KB
[23252:0x102801600] Large object space, used:      0 KB, available: 1446391 KB, committed:      0 KB
[23252:0x102801600] All spaces,         used:   6958 KB, available: 1450743 KB, committed:  17920KB
[23252:0x102801600] External memory reported:     74 KB
[23252:0x102801600] Total time spent in GC  : 8.4 ms
[23252:0x102801600] Fast promotion mode: false survival rate: 1%
