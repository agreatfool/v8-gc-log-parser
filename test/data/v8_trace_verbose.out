[
  {
    "pid": 23252,
    "isolate": "0x102801600",
    "time_since_init": 769,
    "data": {
      "collector": "scavenge",
      "start_object_size": 11.8,
      "end_object_size": 8,
      "start_memory_size": 18.5,
      "end_memory_size": 18.5,
      "pause": 0.5,
      "external_time": 0
    },
    "type": "trace"
  },
  {
    "pid": 23252,
    "isolate": "0x102801600",
    "type": "verbose",
    "data": {
      "memory_allocator": {
        "used": 18944,
        "available": 1447424
      },
      "new_space": {
        "used": 90,
        "available": 3937,
        "committed": 8192
      },
      "old_space": {
        "used": 4983,
        "available": 0,
        "committed": 5120
      },
      "code_space": {
        "used": 2523,
        "available": 3,
        "committed": 3072
      },
      "map_space": {
        "used": 621,
        "available": 0,
        "committed": 1024
      },
      "large_object_space": {
        "used": 0,
        "available": 1446903,
        "committed": 0
      },
      "all_spaces": {
        "used": 8218,
        "available": 1450845,
        "committed": 17408
      },
      "external_memory_reported": 56,
      "total_time_spent_in_gc": 7
    }
  },
  {
    "pid": 23252,
    "isolate": "0x102801600",
    "time_since_init": 921,
    "data": {
      "collector": "scavenge",
      "start_object_size": 12,
      "end_object_size": 8.2,
      "start_memory_size": 19,
      "end_memory_size": 19,
      "pause": 0.6,
      "external_time": 0
    },
    "type": "trace"
  },
  {
    "pid": 23252,
    "isolate": "0x102801600",
    "type": "verbose",
    "data": {
      "memory_allocator": {
        "used": 19456,
        "available": 1446912
      },
      "new_space": {
        "used": 47,
        "available": 3980,
        "committed": 8192
      },
      "old_space": {
        "used": 5110,
        "available": 375,
        "committed": 5632
      },
      "code_space": {
        "used": 2534,
        "available": 3,
        "committed": 3072
      },
      "map_space": {
        "used": 677,
        "available": 0,
        "committed": 1024
      },
      "large_object_space": {
        "used": 0,
        "available": 1446391,
        "committed": 0
      },
      "all_spaces": {
        "used": 8369,
        "available": 1450751,
        "committed": 17920
      },
      "external_memory_reported": 58,
      "total_time_spent_in_gc": 7.6
    }
  },
  {
    "pid": 23252,
    "isolate": "0x102801600",
    "time_since_init": 930,
    "data": {
      "collector": "mark_sweep",
      "start_object_size": 8.4,
      "end_object_size": 6.8,
      "start_memory_size": 19,
      "end_memory_size": 19,
      "pause": 0.8,
      "external_time": 0,
      "step_description": "+ 6.2 ms in 10 steps since start of marking, biggest step 1.0 ms, walltime since start of marking 14 ms"
    },
    "type": "trace"
  },
  {
    "pid": 23252,
    "isolate": "0x102801600",
    "type": "verbose",
    "data": {
      "memory_allocator": {
        "used": 19456,
        "available": 1446912
      },
      "new_space": {
        "used": 145,
        "available": 3882,
        "committed": 8192
      },
      "old_space": {
        "used": 4233,
        "available": 469,
        "committed": 5632
      },
      "code_space": {
        "used": 2287,
        "available": 0,
        "committed": 3072
      },
      "map_space": {
        "used": 292,
        "available": 0,
        "committed": 1024
      },
      "large_object_space": {
        "used": 0,
        "available": 1446391,
        "committed": 0
      },
      "all_spaces": {
        "used": 6958,
        "available": 1450743,
        "committed": 17920
      },
      "external_memory_reported": 74,
      "total_time_spent_in_gc": 8.4
    }
  }
]
